Refactors dockerfile lint worker

navidshaikh opened the pull request:
This changeset refactors dockerfile lint worker to adhere to new
  base worker format we have introduced. Also, the changeset adds
  entries in default_settings.py from constants.py which were
  relevant to lint worker.

navidshaikh committed:
Refactors dockerfile lint worker

  This changeset refactors dockerfile lint worker to adhere to new
  base worker format we have introduced. Also, the changeset adds
  entries in default_settings.py from constants.py which were
  relevant to lint worker.

navidshaikh committed:
Removed unnecessary __init__ method for Dockerfile lint class

navidshaikh commented:
@rtnpro @dharmit @bamachrn @mohammedzee1000 PTAL and review.

rtnpro commented on container_pipeline/workers/dockerfile_lint.py:
```diff
@@ -0,0 +1,131 @@
+        Lint the Dockerfile received
+        """
+        out, err = subprocess.Popen(
+            ["docker",
```
you can use ``run_cmd`` from ``lib/cmd.py``

rtnpro commented on container_pipeline/workers/dockerfile_lint.py:
```diff
@@ -0,0 +1,131 @@
+import json
+import logging
+import os
+import subprocess
```
Use ``container_pipeline.lib.cmd.run_cmd`` to replace direct interactions with subprocess.

rtnpro reviewed (changes requested):
Make the suggested changes and this is good to merge.

dharmit commented on container_pipeline/workers/dockerfile_lint.py:
```diff
@@ -0,0 +1,131 @@
+
+class DockerfileLintWorker(BaseWorker):
+    """
+    Dockerfile lint worker
```
Please don't add a docstring for the sake of it. Explain the behavior or don't add it at all.

dharmit replied:
Here's what I'd like to see:
```python
"""
Dockerfile lint worker lints the Dockerfile using the projectatomic's dockerfile_lint tool.
https://github.com/projectatomic/dockerfile_lint
"""
```

dharmit commented on container_pipeline/workers/dockerfile_lint.py:
```diff
@@ -0,0 +1,131 @@
+            logs_file_path = os.path.join(
+                job.get("logs_dir"),
+                settings.LINTER_RESULT_FILE
+
```
Erroneous newline

dharmit commented on container_pipeline/workers/dockerfile_lint.py:
```diff
@@ -0,0 +1,131 @@
+                "notify_email": job.get("notify_email"),
+                "job_name": job.get("job_name"),
+                "msg": None,
+                "linter_results_path": logs_file_path,
```
We're creating `logs_file_path` above. How about we instead create `linter_results_path`? Suggesting this as there's no modification happening, it would make sense when we assign that to the dictionary key `"linter_results_path"`, and it would be easier to understand for someone reading the code.

dharmit commented on container_pipeline/workers/dockerfile_lint.py:
```diff
@@ -0,0 +1,131 @@
+            settings.LINTER_STATUS_FILE
+        )
+        # now export the status about linter execution in logs dir of the job
+        # this response will be read after job builds and sending email to user
```
Do we mean this - after job builds and ***before*** sending email to user?

navidshaikh replied:
Yes. We read the status file and result file while building the email.

dharmit commented on container_pipeline/workers/dockerfile_lint.py:
```diff
@@ -0,0 +1,131 @@
+        """
+        try:
+            fin = open(status_file_path, "w")
+            json.dump(status, fin)
```
A file opened for any purpose should be closed as well, right? Also, how about using `with open` instead?

navidshaikh replied:
The closing of file will be done when the method returns.

dharmit reviewed (changes requested):
Besides specific comments, please replace the entries for `logger` with `self.logger`. I don't see how that would work otherwise.

navidshaikh committed:
Uses library run_cmd method, removes subprocess imports

navidshaikh committed:
Changes variable names and updated docstrings of class

  Updated docstring of class
  Changed the variable names used
  Opened and exported in file with `with` statement

navidshaikh committed:
Uses the right logging object to log the messages

  Uses self.logger instead of logger for logging messages.

navidshaikh commented:
@dharmit @rtnpro : I have updated the PR per our discussion and review. PTAL

rtnpro commented on container_pipeline/lib/default_settings.py:
```diff
@@ -9,8 +9,18 @@
+LOGS_DIR_BASE = "/srv/pipeline-logs/"
+
+LOGS_URL_BASE = "https://registry.centos.org/pipeline-logs/"
+
```
Unnecessary new line

rtnpro commented on container_pipeline/lib/default_settings.py:
```diff
@@ -9,8 +9,18 @@
 SERVICE_LOGFILE = "service_debug.log"
 
+LINTER_RESULT_FILE = "linter_results.txt"
+
```
New lines are not required after each line. You may want to group multiple lines into logical paragraph and have an empty line between paragraphs.

navidshaikh committed:
Groups variables logically in default settings

 Removes unnecessary newlines and groups the constants together.

navidshaikh commented:
@rtnpro : Updated PR per review comments. PTAL  Cc @dharmit @bamachrn

dharmit commented on container_pipeline/workers/dockerfile_lint.py:
```diff
@@ -0,0 +1,127 @@
+            out = run_cmd(command)
+        except Exception as e:
+            self.logger.error(
+                "Dockerfile Lint check failed", extra={'locals': locals()})
```
Is the `extra={'locals': locals()}` still relevant/required? cc: @rtnpro

navidshaikh replied:
I think yes, its all about logging verbose debug information at the time of exception.

dharmit commented:
LGTM

dharmit reviewed (approved):
LGTM except that one piece that needs to be checked for `extra`.

navidshaikh commented:
@rtnpro : is this PR going to be taken for consolidated refactor of pipeline workers?

rtnpro commented:
Merged manually.
